File: app/src/main/java/com/xephorium/crystalnote/ui/util/DisplayUtils.java
package com.xephorium.crystalnote.ui.util;

import android.app.Activity;
import android.graphics.Point;
import android.graphics.Rect;
import android.view.Display;
import android.view.View;

public class DisplayUtils {

    public static int getAppHeight(Activity activity) {
        Rect rect = new Rect();
        View rootView = activity.getWindow().getDecorView();
        rootView.getWindowVisibleDisplayFrame(rect);
        return rect.bottom - rect.top;
    }

    public static int getAppWidth(Activity activity) {
        Rect rect = new Rect();
        View rootView = activity.getWindow().getDecorView();
        rootView.getWindowVisibleDisplayFrame(rect);
        return rect.right - rect.left;
    }

    public static int getDisplayHeight(Activity activity) {
        Display display = activity.getWindowManager().getDefaultDisplay();
        Point size = new Point();
        display.getSize(size);
        return size.y;
    }

    public static int getDisplayWidth(Activity activity) {
        Display display = activity.getWindowManager().getDefaultDisplay();
        Point size = new Point();
        display.getSize(size);
        return size.x;
    }

    public static int getStatusBarHeight(Activity activity) {
        return activity.getResources().getDimensionPixelSize(
                activity.getResources().getIdentifier("status_bar_height", "dimen", "android"));
    }

    public static int getAbsoluteViewTop(View view) {
        int[] coordinates = new int[2];
        view.getLocationOnScreen(coordinates);
        return coordinates[1];
    }
}

File: app/src/main/java/com/xephorium/crystalnote/ui/utility/DisplayUtils.kt
package com.xephorium.crystalnote.ui.utility

import android.app.Activity
import android.graphics.Point
import android.graphics.Rect
import android.view.View
import androidx.core.content.ContextCompat

object DisplayUtils {


    /*--- Public Utility Methods ---*/

    fun getAppHeight(activity: Activity): Int {
        val rect = Rect()
        val rootView = activity.window.decorView
        rootView.getWindowVisibleDisplayFrame(rect)
        return rect.bottom - rect.top
    }

    fun getAppWidth(activity: Activity): Int {
        val rect = Rect()
        val rootView = activity.window.decorView
        rootView.getWindowVisibleDisplayFrame(rect)
        return rect.right - rect.left
    }

    fun getDisplayHeight(activity: Activity): Int {
        val display = activity.windowManager.defaultDisplay
        val size = Point()
        display.getSize(size)
        return size.y
    }

    fun getDisplayWidth(activity: Activity): Int {
        val display = activity.windowManager.defaultDisplay
        val size = Point()
        display.getSize(size)
        return size.x
    }

    fun getStatusBarHeight(activity: Activity): Int {
        return activity.resources.getDimensionPixelSize(
                activity.resources.getIdentifier("status_bar_height", "dimen", "android"))
    }

    fun getAbsoluteViewTop(view: View): Int {
        val coordinates = IntArray(2)
        view.getLocationOnScreen(coordinates)
        return coordinates[1]
    }
}